crates/sui-checkpoint-object-store-indexer/src/lib.rs

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::{Arc, LazyLock};
+
+use anyhow::Context;
+use bytes::{BufMut, Bytes, BytesMut};
+use object_store::path::Path as ObjectPath;
+use object_store::{Error as ObjectStoreError, PutMode, PutPayload};
+use prost::Message;
+use sui_indexer_alt_framework::pipeline::Processor;
+use sui_indexer_alt_framework::pipeline::concurrent::Handler;
+use sui_indexer_alt_framework::store::StoreTypes;
+use sui_indexer_alt_object_store::ObjectStore;
+use sui_rpc::field::{FieldMask, FieldMaskUtil};
+use sui_rpc::merge::Merge;
+use sui_rpc::proto::sui::rpc;
+use sui_types::full_checkpoint_content::Checkpoint;
+
+pub struct CheckpointBlob {
+    pub sequence_number: u64,
+    pub proto_bytes: Bytes,
+}
+
+pub struct CheckpointBlobPipeline {
+    pub compression_level: Option<i32>,
+}
+
+#[async_trait::async_trait]
+impl Processor for CheckpointBlobPipeline {
+    const NAME: &'static str = "checkpoint_blob";
+    type Value = CheckpointBlob;
+
+    async fn process(&self, checkpoint: &Arc<Checkpoint>) -> anyhow::Result<Vec<Self::Value>> {
+        static MASK: LazyLock<sui_rpc::field::FieldMaskTree> = LazyLock::new(|| {
+            FieldMask::from_paths([
+                rpc::v2::Checkpoint::path_builder().sequence_number(),
+                rpc::v2::Checkpoint::path_builder().summary().bcs().value(),
+                rpc::v2::Checkpoint::path_builder().signature().finish(),
+                rpc::v2::Checkpoint::path_builder().contents().bcs().value(),
+                rpc::v2::Checkpoint::path_builder()
+                    .transactions()
+                    .transaction()
+                    .bcs()
+                    .value(),
+                rpc::v2::Checkpoint::path_builder()
+                    .transactions()
+                    .effects()
+                    .bcs()
+                    .value(),
+                rpc::v2::Checkpoint::path_builder()
+                    .transactions()
+                    .effects()
+                    .unchanged_loaded_runtime_objects()
+                    .finish(),
```

Comment:
For my education, what does `finish` do here? It doesn't look like a field name, and all the other cases seem to not require it.